### PR TITLE
macOS-related: build, test, and trie compare function

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,14 +69,16 @@ make build          # compile and link
...
-  STATIC_LIBSTDCXX=0  # link libstdc++ dynamically (default: 1)
   NO_TESTS=1          # disable unit tests
   WHY=1               # explain CMake decisions (in /tmp/cmake-why)
   FORCE=1             # Force CMake rerun (default)
   CMAKE_ARGS=...      # extra arguments to CMake
   VG=1                # build for Valgrind
   SAN=type            # build with LLVM sanitizer (type=address|memory|leak|thread) 
   SLOW=1              # do not parallelize build (for diagnostics)
+  GCC=1               # build with GCC (default unless Sanitizer)
+  CLANG=1             # build with CLang
```
Comment (oshadmi) [inline Makefile:80]: Do we want to expose `CLANG=1` for non-mac and and support a "build matrix"?

File: src/trie/trie.c [hunk 2]
```diff
@@ -465,6 +454,19 @@ static int runecmp(const rune *sa, size_t na, const rune *sb, size_t nb) {
   return 0;
 }
 
+// comparator for node sorting by child max score
```
Comment (oshadmi) [inline src/trie/trie.c:457]:
```suggestion
// comparator for node sorting by child max score and, if score is equal, by string
```

File: tests/cpptests/test_cpp_index.cpp [hunk 3]
```diff
@@ -1083,7 +1083,11 @@ TEST_F(IndexTest, testHugeSpec) {
   s = IndexSpec_Parse("idx", (const char **)&args[0], args.size(), &err);
   ASSERT_TRUE(s == NULL);
   ASSERT_TRUE(QueryError_HasError(&err));
+#if !defined(__arm__) && !defined(__aarch64__)
   ASSERT_STREQ("Schema is limited to 128 TEXT fields", QueryError_GetError(&err));
+#else
+  ASSERT_STREQ("Schema is limited to 64 TEXT fields", QueryError_GetError(&err));
+#endif
```
Comment (oshadmi) [inline tests/cpptests/test_cpp_index.cpp:1090]: We test other flags in redisearch.h
```
#if defined(__x86_64__) && !defined(RS_NO_U128)
/* 64 bit architectures use 128 bit field masks and up to 128 fields */
typedef __uint128_t t_fieldMask;
#define RS_FIELDMASK_ALL (((__uint128_t)1 << 127) - (__uint128_t)1 + ((__uint128_t)1 << 127))
#else
/* 32 bit architectures use 64 bits and 64 fields only */
typedef uint64_t t_fieldMask;
#define RS_FIELDMASK_ALL 0xFFFFFFFFFFFFFFFF
#endif
```

File: tests/pytests/CMakeLists.txt [hunk 4]
```diff
@@ -37,13 +33,15 @@ foreach(n ${PY_TEST_FILES})
...
 		WORKING_DIRECTORY ${PROJECT_BINARY_DIR})
 
-    add_test(NAME "PY_${test_name}_RAW_DOCID_ENCODING"
-		COMMAND bash -c "MODARGS='RAW_DOCID_ENCODING true;' ${baseCommand} -t ${n}"
-		WORKING_DIRECTORY ${PROJECT_BINARY_DIR})
+	if (NOT USE_COVERAGE)
+		add_test(NAME "PY_${test_name}_RAW_DOCID_ENCODING"
+			COMMAND bash -c "MODARGS='RAW_DOCID_ENCODING true;' ${baseCommand} -t ${n}"
+			WORKING_DIRECTORY ${PROJECT_BINARY_DIR})
+	endif()
```
Comment (oshadmi) [inline tests/pytests/CMakeLists.txt:40]: No need for coverage of `RAW_DOCID_ENCODING`?

**Review by oshadmi — COMMENTED** [review]
Few comments

File: Makefile (hunk 1 above)
Comment (rafie) [reply Makefile:80]: Yes, this is relevant (implicity) to sanitizer, and if you make clang available on your system (e.g. via `readies/bin/getclang`).
I'm considering doing an implicit variant for non-default compiler selections.
PS there's also INTELCC under way.

File: tests/cpptests/test_cpp_index.cpp (hunk 3 above)
Comment (rafie) [reply tests/cpptests/test_cpp_index.cpp:1090]: `RS_NO_U128` is a weirdo. I think we should be more explicit in redisearch.h checking for ARM values.

File: tests/pytests/CMakeLists.txt (hunk 4 above)
Comment (rafie) [reply tests/pytests/CMakeLists.txt:40]: Not sure it worth the time spent there. Ariel recommended we remove it.

File: src/trie/trie.c [hunk 5]
```diff
@@ -465,6 +454,19 @@ static int runecmp(const rune *sa, size_t na, const rune *sb, size_t nb) {
...
+static int __trieNode_Cmp_Score(const void *p1, const void *p2) {
+  TrieNode *n1 = *(TrieNode **)p1;
+  TrieNode *n2 = *(TrieNode **)p2;
+
+  if (n1->maxChildScore < n2->maxChildScore) {
+    return 1;
+  } else if (n1->maxChildScore > n2->maxChildScore) {
+    return -1;
+  }
+  return runecmp(n1->str, n1->len, n2->str, n2->len);
```
Comment (ashtul) [inline src/trie/trie.c:467]: is it enough to compare a single `rune`

**Review by ashtul — COMMENTED** [review]
One small comment

File: src/trie/trie.c (hunk 5 above)
Comment (ashtul) [inline src/trie/trie.c:467]:
```suggestion
  return __trieNode_Cmp_Lex(p1, p2);
```

Comment (ashtul) [reply src/trie/trie.c:467]: I see we do the same on `Cmp_Lex`. ignore.

File: src/trie/trie.c [hunk 6]
```diff
@@ -465,6 +454,19 @@ static int runecmp(const rune *sa, size_t na, const rune *sb, size_t nb) {
...
+}
+
 static int __trieNode_Cmp_Lex(const void *a, const void *b) {
   const TrieNode *na = *(const TrieNode **)a, *nb = *(const TrieNode **)b;
   return runecmp(na->str, na->len, nb->str, nb->len);
```
Comment (ashtul) [inline src/trie/trie.c:472]:
```suggestion
  return na->str[0] - nb->str[0];
```

Comment (rafie) [reply src/trie/trie.c]: We'll do this in a subsequent PR.

**Review by ashtul — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (oshadmi) [reply Makefile:80]: :astonished:

**Review by oshadmi — APPROVED** [review]

**Pull request merged**